### PR TITLE
Prevent errant HTML from crashing server islands

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Closes https://github.com/withastro/astro/issues/11638
- Closes https://github.com/withastro/astro/issues/11597
- Prevents infinite loop when fallback comment is missing.

## Testing

- Test added, doesn't crash

## Docs

N/A, bug fix

### CODE REVIEW

**Comment by Princesseuh** [comment]
Should we consider another way of doing this to prevent this error from occurring?

**Comment by ematipico** [comment]
@matthewp can you provide some context about the issue and how you fixed it? 

I read the code and I can't understand:
- what's an errant HTML
- how `HTMLError` fixes the issue

File: packages/astro/src/runtime/server/render/server-islands.ts [hunk 1]
```diff
@@ -85,9 +85,10 @@ if(response.status === 200 && response.headers.get('content-type') === 'text/htm
 	let html = await response.text();
 
 	// Swap!
-	while(script.previousSibling?.nodeType !== 8 &&
-		script.previousSibling?.data !== 'server-island-start') {
-		script.previousSibling?.remove();
+	while(script.previousSibling &&
+		script.previousSibling.nodeType !== 8 &&
```
[the remark this answers is not in the document]
Comment (matthewp) [reply packages/astro/src/runtime/server/render/server-islands.ts:89]: Not worth the extra bytes imo, these are well-known constants, I reference this when I forget them: https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType

**Comment by matthewp** [comment]
> Should we consider another way of doing this to prevent this error from occurring?

Sure, do you have any ideas? Using comments as markers for chunks of HTML fragments is pretty common in frameworks, Solid uses this technique for their SSR for ex. Cloudflare shouldn't be doing this. So the goal of this PR was simply to prevent worst-case of locking up the browser.

**Comment by Princesseuh** [comment]
I don't have any ideas that isn't massively over-engineered, ha. I agree that it's a bit odd, but I doubt it's the only minifier that does this.

**Review by Princesseuh — APPROVED** [review]

**Comment by matthewp** [comment]
This happens in SolidJS as well: https://github.com/solidjs/solid/issues/1697

So I think we should go with this solution as there's no real way to do this without comments that's reasonable.

**Review by bluwy — APPROVED** [review]
also fwiw svelte 5 uses comments as hydration markers

**Review by ematipico — APPROVED** [review]

**Pull request merged**